Headers/DebugServer2/Types.h

```diff
@@ -239,7 +239,7 @@ inline BOOL AllocateAndCopySid(const PSID pSid, PSID *ppSid) noexcept {
   if (!IsValidSid(pSid))
     return CreateWellKnownSid(WinNullSid, nullptr, ppSid, nullptr);
 
-  DWORD nSubAuthorityCount = *GetSidSubAuthorityCount(pSid);
+  UCHAR nSubAuthorityCount = *GetSidSubAuthorityCount(pSid);
```

Comment:
PR description says PUCHAR but this says UCHAR. Intentional?

Reply:
Yes, there is a de-reference on the return value, which changes the `PUCHAR` to `UCHAR`.  The subsequent uses are for using the `PUCHAR` as `CHAR`/`UCHAR`, so it makes sense to immediately dereference.  We cannot do anything protective here as `GetSidSubAuthorityCount` explicitly states that any failure is UB.